### PR TITLE
Adjust linters.

### PR DESCRIPTION
**Pull request by sherakama**
# READY

# Summary
- Our codeclimate score suffered when we moved the code around as all the old issues came back.
- Looking at the documentation it appears that codeclimate was not respecting our lint configuration file. We had `scss-lint` enabled in codeclimate but we had configuration for `sass-lint` in our repo. They are two different things and need to be one.
- I revised the sass-lint.yml to cut down on errors and have made the first pass at clean up. I think I have gotten it down to real issues we should fix now.
- The linter picked up a few bugs along the way. Should be fixed now.
- Also, turn off markdown linting cuz no1curr.

# Needed By (Date)
- Meh

# Urgency
- Low

# Steps to Test

1. Review documentation
2. Review changes to the sass-lint.yml file
3. Review code changes
4. Run a `npm run build` 
5. Run a `npm run sasslint`
6. Leave comments and make changes. 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
Sorry, this is not ready. Needs tweaks.

**Comment by sherakama** [comment]
@josephgknox @JBCSU @yvonnetangsu 

So I finally figured out why our sass lint settings were different in our code climate results. We were using the wrong linter 🤦‍♂️. This PR revises the linting configuration and takes a first comb through the issues to boil them down to 'real' ones.

File: .sass-lint.yml [hunk 1]
```diff
@@ -2,57 +2,58 @@
...
       max-depth: 4
+  no-color-literals: 0
+  no-css-comments: 0
+  no-empty-rulesets: 1
+  no-ids: 1
+  no-mergeable-selectors: 0
+  no-misspelled-properties: 1
+  no-vendor-prefixes: 0
+  no-url-domains: 0
+  no-warn: 0
```
Comment (yvonnetangsu) [inline .sass-lint.yml]:
```suggestion
  no-warn: 0
    no-qualifying-elements:
    - 1
    -
      allow-element-with-attribute: 1
```

Comment (yvonnetangsu) [reply .sass-lint.yml]: I also pushed up a small fix in CTA after the new linter found a small issue. Linter is working!

**Comment by JBCSU** [comment]
@sherakama I'm not sure I understand how to test this PR. I followed the instructions

- `npm run build`
- `npm run sasslint`

and I got 1131 warnings, including tons of warnings about whitespace and semicolons. When I look at the codeclimate details there are 7 issues that aren't marked **Fixed**, none of which are about whitespace or semicolons. So it seems to me that the lint configurations don't match. What am I looking for in order to validate this PR?

**Comment by JBCSU** [comment]
FYI, CC's issue with not allowing filename extensions on `@import` statements is referring to the `@import bourbon.scss` directives in our 3 rollup .scss files. The `.scss` extension is required to work around a bug in webpack's sass-loader: bourbon has both a `bourbon.js` and `bourbon.scss` rollup. When sass-loader is processing a `.scss` file with `@import bourbon`, and both a `.js` and a `.scss` file could resolve, it incorrectly tries to import the `.js` file. We should keep an eye on the sass-loader package. When that bug is fixed we can remove the `.scss` extension from our `@import`s.

**Comment by yvonnetangsu** [comment]
> @sherakama I'm not sure I understand how to test this PR. I followed the instructions
> 
> * `npm run build`
> * `npm run sasslint`
> 
> and I got 1131 warnings, including tons of warnings about whitespace and semicolons. When I look at the codeclimate details there are 7 issues that aren't marked **Fixed**, none of which are about whitespace or semicolons. So it seems to me that the lint configurations don't match. What am I looking for in order to validate this PR?

Weird....when I ran it last time, I got 0 error and 2x warnings 🤔

**Comment by josephgknox** [comment]
> > @sherakama I'm not sure I understand how to test this PR. I followed the instructions
> > 
> > * `npm run build`
> > * `npm run sasslint`
> > 
> > and I got 1131 warnings, including tons of warnings about whitespace and semicolons. When I look at the codeclimate details there are 7 issues that aren't marked **Fixed**, none of which are about whitespace or semicolons. So it seems to me that the lint configurations don't match. What am I looking for in order to validate this PR?
> 
> Weird....when I ran it last time, I got 0 error and 2x warnings 🤔

+1. 🤔

**Comment by JBCSU** [comment]
Y'know what I should probably do? I should probably test your branch, instead of `master`. That might help.

D'oh! :woman_facepalming: Sorry 'bout that. My brain's fried after 2 days training.

**Comment by JBCSU** [comment]
When I test the correct branch, 0 errors, 28 warnings. That's much mo' bettah. Thanks for doing this. It make linting actually useful.

**Pull request merged**